UMprod

jackeown opened the pull request:
It looks like these can be automatically merged. 
This is the latest umiami production version (merged with your latest changes)
Maybe there's something we need to do to flatten the commit history, but it should all work :)

Let me or Geoff know if there's something wrong with this.

Thanks,
Jack

jackeown committed:
old changes from Rahul

jackeown committed:
old changes from Rahul

jackeown committed:
Removed almost all log.info statements.

jackeown committed:
Removed almost all log.info statements.

Commit:
some recycle bin renaming

Commit:
some recycle bin renaming

Commit:
changed starlogo.png

Commit:
Merge branch 'UMprod' of https://github.com/jackeown/StarExec into recycleBinRename

Commit:
removed copyright notice

Commit:
Updated RunSolver binary and added its source code to the repository

Commit:
finished recycle bin rename in interface

Commit:
Merge branch 'recycleBinRename' into UMprod

Commit:
recycle bin rename fix

Commit:
Merge branch 'recycleBinRename' into UMprod

Commit:
I had missed some renames

Commit:
forgot more recycle renaming

Commit:
Merge branch 'recycleBinRename' into UMprod

Commit:
more recycle bin renaming

Commit:
changed /starexec/secure/explore/spaces.jsp to have consistent confirmations for removing jobs/solvers/benchmarks etc...

Commit:
Merge branch 'recycleBinRename' into UMprod

Commit:
made dialog boxes bigger

Commit:
Merge branch 'recycleBinRename' into UMprod

Commit:
first approximation of what Terms of Service will look like

jackeown committed:
test a few changes to the interface for removing things from spaces.

jackeown committed:
test a few changes to the interface for removing things from spaces.

jackeown committed:
test a few changes to the interface for removing things from spaces.

jackeown committed:
more of the same

jackeown committed:
more of the same

jackeown committed:
renamed recycle to trash more

jackeown committed:
Merge branch 'UMdev' of https://github.com/jackeown/StarExec into UMdev

jackeown committed:
fixed 'All' and 'None' for selecting benchmarks/solvers/etc. and added shift for selecting ranges of rows.

jackeown committed:
fixed 'All' and 'None' for selecting benchmarks/solvers/etc. and added shift for selecting ranges of rows.

jackeown committed:
Merge branch 'UMdev' of https://github.com/jackeown/StarExec into UMdev

jackeown committed:
Merge branch 'UMdev' of https://github.com/jackeown/StarExec into UMdev

jackeown committed:
stupid merge crap

jackeown committed:
Merge branch 'UMdev' into recycleBinRename

jackeown committed:
Revert "renamed recycle to trash more"

This reverts commit e4703417e80e03285aa0278afa94c376f07a53c1.

jackeown committed:
Made it so you can move solvers to trash in hierarchy.

jackeown committed:
Merge branch 'UMdev' of https://github.com/jackeown/StarExec into UMdev

jackeown committed:
terms of service update

jackeown committed:
Merge branch 'UMdev' of https://github.com/jackeown/StarExec into UMdev

jackeown committed:
made shift-select a little better...still terrible...but better. Should use dataTable.select in the future.

jackeown committed:
added IDV (hopefully)

jackeown committed:
added GDV (hopefully)

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
updating GDV

jackeown committed:
added video folder to public

jackeown committed:
Merge branch 'fb1' into UMdev

jackeown committed:
disabled graphs on job info page

jackeown committed:
disabled more stuff in job details page...perhaps will break

jackeown committed:
disabled more stuff in job details page...perhaps will break...again

jackeown committed:
added debugging info

jackeown committed:
added more logging to UploadSolver.java

mxzahid committed:
added chown to util.java

mxzahid committed:
.

mxzahid committed:
.

mxzahid committed:
.

mxzahid committed:
.

mxzahid committed:
.

mxzahid committed:
.

mxzahid committed:
.

mxzahid committed:
debugging permissions

mxzahid committed:
chown remove

mxzahid committed:
updated MinimalData.sql for passwords

jackeown committed:
changed more 'recycled' to 'trashed'

jackeown committed:
changed 'trashed' to 'in trash'

jackeown committed:
fixed weird jstree border glitch

jackeown committed:
fixed weird jstree border glitch...for real

mxzahid committed:
logging for weird sanbox dir

mxzahid committed:
adds write permission to sandbox

mxzahid committed:
fsdfs

mxzahid committed:
removed chmod command for weird directory

mxzahid committed:
readds chmod

astump committed:
small tweak to overcome issue copying files to sandbox directory

New temporary directories seem to get created without group write
permission as of the tc9 upgrade (based on comments from Alex in the
code), and the code is counting on having that permission.  Simple
solution is to chmod to add group-write permission.  Alex already had
a function in there to do that, so we just call it.

jackeown committed:
Merge branch 'fb1' of https://github.com/StarExec/StarExec into UMdev

jackeown commented:
It looks like this also includes our change of the logo color though...How do I remove a specific change from a pull request?

astump commented:
I am sorry, but I do not think I will be able to accept this large PR.  It would take a lot of time to review and test on my end.  I hope you understand that as it stands, I have almost no incentive to undertake that work.  I do not know at a high level what changes you have made, or the risks that they might cause problems on StarExec Iowa.  At the best, the outcome will be that the two sites are in synch.  But you are not really making any pitch for some bug fixes or new features or things that I might want to adopt here.  I just have to be candid about this.  If you have some more details you want to share about this PR and why it could be good to incorporate, please share them.

jackeown committed:
Geoff has power

aarijcarcade committed:
Geoff has more power

aarijcarcade committed:
Geoff eats cherrys

aarijcarcade committed:
Recovering

aarijcarcade committed:
Recovered at last

aarijcarcade committed:
The last recovery

aarijcarcade committed:
Added back Jack's destructive rm

aarijcarcade committed:
Added Cluster.MachineSpecs.txt and overrides.properties so that build is reproducable via git clone -> ant build -> script/soft-deploy.sh

aarijcarcade committed:
FIxed web.xml

aarijcarcade committed:
Merge branch 'UMprod' of github.com:StarExecMiami/StarExec into UMprod

jackeown committed:
Update Cluster.MachineSpecs.txt

aarijcarcade committed:
Cleansing, now using Starexec-deploy

aarijcarcade committed:
Merge branch 'UMprod' of github.com:StarExecMiami/StarExec into UMprod

aarijcarcade committed:
cleaned sge and xcat

aarijcarcade committed:
updated index.html

jackeown committed:
Merge remote-tracking branch 'origin/AZ' into UMprod

jackeown committed:
test

jackeown committed:
test

jackeown committed:
Update functions.bash

jackeown committed:
Merge remote-tracking branch 'upstream/fb1' into pullingUpdates

jackeown committed:
changed runsolver makefile to make it compile...

jackeown committed:
Updated runsolver

jackeown committed:
Updated runsolver

jackeown committed:
Updated runsolver

jackeown committed:
Updated runsolver

jackeown committed:
Update LocalBackend.java

jackeown committed:
added KubernettesBackend and made it submit jobs quicker (decreased periodic task from every 60sec to every 20sec

jackeown committed:
made it compile hopefully...

jackeown committed:
changed local backend

jackeown committed:
testing run_image in starexec with dummy prover

jackeown committed:
testing run_image in starexec with dummy prover

jackeown committed:
testing run_image in starexec with dummy prover

jackeown committed:
added k8s support hopefully

jackeown committed:
added k8s support hopefully

jackeown committed:
added k8s support hopefully

jackeown committed:
added k8s support hopefully

jackeown committed:
added k8s support hopefully

jackeown committed:
Add logging to verify this is where the jobscript copies the solver output back to the head node...

jackeown committed:
Update jobscript

jackeown committed:
Update functions.bash (moved copy log before copy command)

jackeown committed:
Update footer.tag (removed copyright message)

jackeown committed:
Update functions.bash (added adjustForK8s function)

jackeown committed:
Update jobscript (k8s stuff)

jackeown committed:
Update functions.bash

jackeown committed:
Update jobscript

jackeown committed:
Update functions.bash

jackeown committed:
Update functions.bash

jackeown committed:
Update R.java (Added k8s backend)

jackeown committed:
renamed k8s backend

jackeown committed:
updated k8s backend (basically local, but now with ~threads~)

jackeown committed:
I forgot a break in a switch statement like an undergrad - ugh

jackeown committed:
Merged PullingUpdates into UMprod

BananaApache committed:
changed default subspace options to no

geoffgeoffgeoff3 committed:
Merge pull request #1 from BananaApache/UMprod

changed default subspace options to no

AndresCdo committed:
fix: remove all user files on deletion

Fixes a filesystem storage leak by ensuring all user-related files are removed when a user is deleted.

Adds comprehensive cleanup of solver, benchmark, and job directories plus deletion of original and thumbnail pictures for users, solvers, and benchmarks. Improves logging of deletion actions and reports freed space; adds basic error handling for picture cleanup to avoid orphaned files and reclaim disk usage.

geoffgeoffgeoff3 committed:
Merge pull request #18 from AndresCdo/feature/delete-user-data

Fix user deletion to remove all related files

AndresCdo committed:
fix: remove personal space when deleting user

Adds logic to locate a user's personal space (identified by firstname_lastname in lowercase) and remove it as part of user deletion. Uses a lookup that scans the user's spaces for the expected name, logs actions and warnings, and continues with user removal even if personal-space deletion fails. Prevents orphaned personal spaces and fixes user-deletion edge cases where a personal space remained behind.

geoffgeoffgeoff3 committed:
Merge pull request #19 from AndresCdo/feature/delete-user-data

Fix user deletion to remove orphaned personal spaces

AndresCdo committed:
fix(users): correct database deletion and prevent inconsistent state on user deletion

Fix critical bug where DeleteUser stored procedure wasn't executing due to incorrect JDBC call (executeQuery vs executeUpdate). This caused deleted users to remain in the database and Admin user list despite filesystem cleanup.

Key changes:
- Replace executeQuery() with executeUpdate() for DeleteUser stored procedure
- Reverse deletion order: database first, then filesystem (prevents orphaned files)
- Preload user entities (jobs, solvers, benchmarks) before any deletion
- Refactor helper methods to accept preloaded lists instead of re-querying DB
- Fix typo in error log tag ("deleteUse" → "deleteUser")

Benefits:
- Deleted users now properly removed from database and Admin user list
- No inconsistent state if DB deletion fails (filesystem untouched)
- More efficient (eliminates redundant database queries during cleanup)
- Safer error handling with early exit on DB failure

geoffgeoffgeoff3 committed:
Merge pull request #22 from AndresCdo/fix/user-deletion

Fix user deletion process to prevent inconsistent state

AndresCdo committed:
feat: add isAdmin caching and simplify user deletion

Adds a short-lived in-memory cache for isAdmin checks to reduce frequent database lookups and improve performance. Invalidates cache entries when a user is deleted or their role is changed.

Refactors user deletion and related helpers to simplify flow and reduce preloading:
- Consolidates directory/picture cleanup helpers to fetch necessary entities on demand instead of passing preloaded lists.
- Streamlines logging and safe-deletion of solver, benchmark, job, and picture directories.
- Ensures cache invalidation after deletion or role updates.

These changes reduce DB load for repeated admin checks and simplify deletion code paths while keeping cleanup behavior intact.

geoffgeoffgeoff3 committed:
Merge pull request #23 from AndresCdo/fix/user-deletion

Fix user deletion process and improve admin checks

AndresCdo committed:
feat(users): add Benchmark and Solver imports for user-related functionality

geoffgeoffgeoff3 committed:
Merge pull request #24 from AndresCdo/fix/user-deletion

Add Benchmark and Solver imports for user functionality

AndresCdo committed:
fix: normalize node names in GridEngine backend

Normalize node names by taking first part before dot (handle FQDNs) in getWorkerNodes and getNodeQueueAssociations methods. This ensures consistency between node names returned by backend and those stored in the database.

AndresCdo committed:
Merge pull request #52 from StarExecMiami/normalize-node-names

fix: normalize node names in GridEngine backend

AndresCdo committed:
fix(sql): refine job pair terminal status codes and completion logic

- Updated `UpdatePairStatus` in JobPairs.sql to include additional terminal status codes (21, 23-26) in the completion check.
- Refined the "job done" check to explicitly look for non-terminal statuses (Pending, Enqueued, Running, Processing, Paused, Awaiting post-processor).
- Updated `GetJobPairOverview` in Jobs.sql to sync pending and error pair counts with the expanded status code list.
- Centralized job pair status updates by making `UpdateJobPairStatus` call `UpdatePairStatus`.

AndresCdo committed:
Merge pull request #54 from StarExecMiami/fix/job-completion-logic-clean

AndresCdo committed:
Fix: Reorder user deletion to prevent orphan permissions

The user deletion process had a critical bug where:
1. Filesystem was deleted before database operations
2. Personal space deletion failures were only warned, not blocking
3. This could leave orphaned permissions if deletion failed mid-process

Changes:
- Move personal space deletion to first step (cleans up permissions)
- Return false and abort if personal space deletion fails
- Move filesystem cleanup to after successful database deletion
- Add detailed comments explaining the order

This matches the approach in the containerised branch and prevents
the scenario where a user loses permissions but account data remains.

AndresCdo committed:
Merge pull request #55 from StarExecMiami/fix/user-deletion-permissions-cleanup

aarijcarcade committed:
updated override.properties to use 10.10.2.254

aarijcarcade committed:
updated override.properties to use 10.10.2.254

aarijcarcade committed:
added web-docs

aarijcarcade committed:
Merge branch 'UMprod' of github.com:StarExecMiami/StarExec into UMprod

aarijcarcade committed:
added runsolve and not passwords

AndresCdo committed:
chore: Add ON DELETE actions to foreign key constraints

Adds ON DELETE CASCADE to several foreign key constraints to ensure referential integrity when parent records are deleted. Changes include cascading deletes for runscript errors, analytics data, and user-event associations, while restricting deletion for processor syntax references to prevent orphaned records.

This improves database consistency and prevents constraint violations during cleanup operations.

AndresCdo committed:
fix(cluster): Correct GetPairsRunningOnNode to show running jobs
Remove reference to deleted status code 3, allowing job pairs with
status_code = 4 (STATUS_RUNNING) to display in cluster node view.

AndresCdo committed:
Merge pull request #60 from StarExecMiami/fix/sql-fk-delete-behavior

AndresCdo committed:
Merge pull request #61 from StarExecMiami/fix/cluster-node-running-jobs

AndresCdo committed:
fix(auth): block anonymous sequential detail access

AndresCdo committed:
fix(auth): harden anonymous download access

Reject mixed anonId/id download requests and validate anonymous links by UUID/type before serving objects.

Persist Tomcat sessions across redeploys; operators must restrict the session store directory before relying on FileStore in production.

AndresCdo committed:
Merge pull request #96 from StarExecMiami/hotfix/umprod-idor-details

fix(auth): block anonymous sequential detail access

AndresCdo committed:
fix(auth): extend WEB-H-01 IDOR auth guards

Extends requiresAuthenticatedPrincipal() in SessionFilter to cover the remaining Mandiant-confirmed protected details, explore, and download resources, plus same-pattern sensitive /secure detail pages found during the local sweep. Existing anonymous-link flows for job, solver, and benchmark pages are preserved.\n\nAdds an in-method authentication guard to the REST reports endpoint (/services/reports/past/) so unauthenticated requests return HTTP 403.\n\nDownload servlet permission failures already map to HTTP 403; malformed requests remain HTTP 400.\n\nMandiant finding: WEB-H-01 (Partially Remediated -> Fully Remediated)

AndresCdo committed:
Merge pull request #99 from StarExecMiami/fix/umprod-idor-web-h-01

fix(auth): extend WEB-H-01 IDOR auth guards
